pilot/pkg/model/context.go

```diff
@@ -154,6 +232,17 @@ func (e *Environment) SetLedger(l ledger.Ledger) {
 	e.ledger = l
 }
 
+// only user for testing
+func (e *Environment) SetClusterLocalHosts(hostList []string) {
+	clusterLocalHosts := make([]host.Name, len(hostList))
+	for _, hostName := range hostList {
+		clusterLocalHosts = append(clusterLocalHosts, host.Name(hostName))
+	}
+	e.mutex.Lock()
+	defer e.mutex.Lock()
```

Comment:
same here.

Reply:
oh sorry ... that as a slight perf improvement, you should manually unlock at the end rather than using `defer`.

Reply:
Hmm, TIL. I thought that was the preferred way of handling unlocks in Go - will fix it.

Reply:
It is, if the function is long and/or has multiple returns. The downside is that there's a slight performance cost. So if it's simple or if the method is called frequently, prefer to explicitly unlock.